### PR TITLE
Default to CAN software kernel timestamp and add config parameter

### PR DESCRIPTION
**Pull request by ruhland**
## Problem
In the cloud logs this error could be seen:
```
{
    "message": "The record timestamp is outside the time range [2022-09-30T05:56:29.218Z, 2022-09-30T09:36:29.218Z) of the memory store",
    "timestamp": 1664518021,
    "eventType": "TIMESTREAM_WRITE_ERROR",
    "logLevel": "ERROR",
    "vehicleName": "fwdemo-215",
    "signalName": "VehicleSpeed",
    "signalValue": 75.2
    "signalTimestampEpochMs": 11300,
}
```
The root cause was that the hardware timestamp are not unix epoch timestmaps ( 11300 would be in 1970).
But the aws-iot-fleetwise-edge agent per default took the hardware timestamp leading to this problem.

## Fix
Per default take the kernel software timestamp and offer a config parameter to switch between the timestamp being used without recompiling.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/vehiclenetwork/src/CANDataSource.cpp [hunk 1]
```diff
@@ -177,6 +177,48 @@ CANDataSource::shouldSleep() const
     return mShouldSleep.load( std::memory_order_relaxed );
 }
 
+Timestamp
+CANDataSource::extractTimestramp( struct msghdr *msgHeader )
```
Comment (gqmelo) [inline src/vehiclenetwork/src/CANDataSource.cpp:181]: extractTimestramp -> extractTimestamp

Comment (gqmelo) [inline src/vehiclenetwork/src/CANDataSource.cpp:181]:
```suggestion
CANDataSource::extractTimestamp( struct msghdr *msgHeader )
```

File: src/vehiclenetwork/src/CANDataSource.cpp [hunk 2]
```diff
@@ -227,44 +269,10 @@ CANDataSource::doWork( void *data )
...
-                }
-                TraceModule::get().setVariable( TraceVariable::MAX_SYSTEMTIME_KERNELTIME_DIFF,
-                                                static_cast<uint64_t>( dataSource->mClock->timeSinceEpochMs() ) -
-                                                    static_cast<uint64_t>( timestamp ) );
-            }
-            else
-            {
-                timestamp = dataSource->mClock->timeSinceEpochMs();
-            }
+            Timestamp timestamp = dataSource->extractTimestramp( &msg[i].msg_hdr );
```
Comment (gqmelo) [inline src/vehiclenetwork/src/CANDataSource.cpp:275]:
```suggestion
            Timestamp timestamp = dataSource->extractTimestamp( &msg[i].msg_hdr );
```

File: src/vehiclenetwork/include/businterfaces/CANDataSource.h [hunk 3]
```diff
@@ -81,6 +115,8 @@ class CANDataSource : public AbstractVehicleDataSource
     // Current non deterministic size of the circular buffer
     size_t queueSize() const;
 
+    Timestamp extractTimestramp( struct msghdr *msgHeader );
```
Comment (gqmelo) [inline src/vehiclenetwork/include/businterfaces/CANDataSource.h:118]:
```suggestion
    Timestamp extractTimestamp( struct msghdr *msgHeader );
```

Comment (ruhland) [reply src/vehiclenetwork/include/businterfaces/CANDataSource.h]: thanks, i missed that, fixed.

**Review by gqmelo — APPROVED** [review]

**Pull request merged**